Jenkinsfile

```diff
@@ -9,32 +9,42 @@ stage('Initialize') {
 
 provision(devEnvironmentName())
 if (isMaster()) {
-    input 'Deploy to Production'
     provision("prod")
 }
 
 def provision(environment) {
-    stage('Provision Infrastructure Stack') {
+    def label = ((environment.startsWith("dev")) ? "dev" : environment).
+        toUpperCase();
+
+    if (label != "DEV") {
+        stage("master") {
+            timeout(time:5, unit:'DAYS') {
+                input "Deploy to ${label}?"
```

Comment:
yep, input should be outside a node, otherwise the node will remain running indefinitely